README.md

```diff
@@ -133,6 +137,75 @@ Assets and events will ignore existing ids on deletes. If you prefer to abort th
 when an attempt to delete an unknown id, use `.option("ignoreUnknownIds", "true")`
 for those resources types.
 
+## Asset hierarchy builder (beta)
+Note: The asset hierarchy builder is currently in beta, and has not been sufficiently tested to be used 
+on production data.
+
+The `.option("type", "assethierarchy")` lets you write new asset hierarchies, or update existing ones,
+using the Spark Data Source. It currently supports writing a single root asset and any number of children.
+The asset hierarchy builder can ingest entire hierarchies, as long as all nodes are connected to the root asset
+through the `externalId`/`parentExternalId` relationship. If input contains an update to data that already exists,
+i.e there's a match on `externalId` and there's a change to one of the other fields, the asset will be updated.
+There's also an option to delete assets from CDF that are not referenced in the input data.
+
+### Requirements
+- The source data must contain a single root, denoted by setting its `parentExternalId` to the empty string `""`.
+- All assets, except root, must be connected to another asset in the input data by `parentExternalId`.
+- The input data must not have loops, to ensure all asset hierarchies are fully connected.
+- `externalId` can not be the empty string `""`.
+
+### Options
+|    Option                  | Default | Description |
+| -------------------------- | --------|------------------------------------------------------------------------------------------------------------ |
+| `deleteMissingAssets`      | `false` | Whether or not you would like assets under the root to be deleted if they're not present in the input data. |
+| `ignoreDisconnectedAssets` | `false` | This will ignore assets that are not connected to root, as opposed to throw an error and stop the program.  |
+| `batchSize`                | 1000    | The number of assets to write per API call.                                                                 |
+
+### Example
+```scala
+val assetHierarchySchema = Seq("externalId", "parentExternalId", "description", "name", "metadata")
+
+// Manually create some assets that satisfy the requirements of the asset hierarchy builder
+val assetHierarchy = Seq(
+  Seq("root_asset", "", "manual_input", "root_asset", Some("This is the root asset"), None),
+  Seq("first_child", "root_asset", "manual_input", "first_child", Some("This is the first_child"), None),
+  Seq("second_child", "root_asset", "manual_input", "second_child", Some("This is the second_child"), None),
+  Seq("grandchild", "", "manual_input", "first_child", Some("This is the root asset"), None)
```

Comment:
Shouldn't this have `first_child` or `second_child` as its `parentExternalId`? Seems like we have two root assets here

Reply:
You're correct, I see I changed the `source` instead of the `parentExternalId`. I'll run the code to verify it.